backend/inspirehep/records/api/base.py

```diff
@@ -553,3 +554,10 @@ def redirected_record_ref(self):
         if self.get("deleted") and pid.status == PIDStatus.REDIRECTED:
             redirection = InspireRedirect.get_redirect(pid)
             return get_ref_from_pid(redirection.pid_type, redirection.pid_value)
+
+    def get_new_deleted_records(self):
+        current_deleted_records = get_value(self, "deleted_records.$ref", [])
+        if current_deleted_records != get_value(
+            self._previous_version, "deleted_records.$ref", []
+        ):
+            return current_deleted_records[-1]
```

Comment:
no, you don't know whether the new one is the last one. You need to compare the two sets.